app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -288,10 +340,20 @@ def add_instance_disk(disks, size, name, location)
         super(disks, size, location, name, "azure")
       end
 
+      def get_security_groups(hardware_network_info)
+        security_group_array = hardware_network_info.collect { |info| info.delete(:security_group) }.compact
+        hardware_network_info.delete_if { |e| e.kind_of?(Hash) && e.empty? }
+
```

Comment:
Because it's meant as a temporary key. Because I've called .delete, the hashes are now empty. Leaving an empty hash caused the weird rspec failures.